mempool/mempool.go

```diff
@@ -256,12 +266,33 @@ func (mem *Mempool) resCbRecheck(req *abci.Request, res *abci.Response) {
 			// Done!
 			atomic.StoreInt32(&mem.rechecking, 0)
 			mem.logger.Info("Done rechecking txs")
+
+			mem.notifyIfTxsAvailable()
 		}
 	default:
 		// ignore other messages
 	}
 }
 
+// TxsAvailable returns a channel which fires once for every height,
+// and only when transactions are available in the mempool.
+// NOTE: the returned channel may be nil if EnableTxsAvailable was not called.
+func (mem *Mempool) TxsAvailable() chan int {
+	return mem.txsAvailable
+}
+
+func (mem *Mempool) notifyIfTxsAvailable() {
+	if mem.Size() == 0 {
+		panic("notified txs available but mempool is empty!")
+	}
+	if mem.txsAvailable != nil &&
+		!mem.notifiedTxsAvailable {
+
+		mem.notifiedTxsAvailable = true
+		mem.txsAvailable <- mem.height + 1
```

Comment:
yes. if we just finished everything to commit block N, then `consensus.height=N+1` since its working on the next height but `mempool.height=N`. So this is the mempool saying it has txs for the next height